test/powershell/engine/Formatting/PSStyle.Tests.ps1

```diff
@@ -286,6 +286,77 @@ Describe 'Tests for $PSStyle automatic variable' {
         $strDec.ContentLength | Should -Be $word.Length
         $strDec.ToString("PlainText") | Should -Be $word
     }
+
+    It "String intput to Out-String should be intact with OutputRendering='<OutputRendering>'" -TestCases @(
+        @{ OutputRendering = 'Ansi'; ContainsAnsi = $true }
+        @{ OutputRendering = 'Host'; ContainsAnsi = $false }
+        @{ OutputRendering = 'PlainText'; ContainsAnsi = $false }
+    ) {
+        param($OutputRendering, $ContainsAnsi)
+
+        $oldRender = $PSStyle.OutputRendering
+        $testStr = "`e[31mABC`e[0m"
+
+        try {
+            $PSStyle.OutputRendering = $OutputRendering
+            ## For input that actually goes through formatting, Out-String should remove VT sequences
+            ## from the formatting output based on the output rendering option that is in effect.
+            (Get-Verb -Verb Get | Out-String).Contains("`e[") | Should -Be $ContainsAnsi
+            ## For string input, since no formatting is applied, Out-String should keep the string intact.
+            ($testStr | Out-String).Trim() | Should -BeExactly $testStr
+        }
+        finally {
+            $PSStyle.OutputRendering = $oldRender
+        }
+    }
+
+    It "String input to Out-File should be intact with OutputRendering='<OutputRendering>'" -TestCases @(
+        @{ OutputRendering = 'Ansi'; }
+        @{ OutputRendering = 'Host'; }
+        @{ OutputRendering = 'PlainText'; }
+    ) {
+        param($OutputRendering)
+
+        $oldRender = $PSStyle.OutputRendering
+        $content = "Read-Host -Prompt '`e[33mEnter your device code`e[0m'"
+        Set-Content -Path $TestDrive\test.ps1 -Value $content -Encoding utf8NoBOM
+
+        try {
+            $PSStyle.OutputRendering = $OutputRendering
+            Get-Content $TestDrive\test.ps1 > $TestDrive\copy.ps1
+            (Get-Content $TestDrive\copy.ps1 -Raw).Trim() | Should -BeExactly $content
+        }
+        finally {
+            $PSStyle.OutputRendering = $oldRender
+            Remove-Item $TestDrive\test.ps1 -Force
```

Comment:
these `Remove-Item` should not be needed. Files in `$TestDrive` are cleaned up by Pester automatically.

Reply:
IIRC, Pester remove the whole `Test` drive after a `describe` finishes, but not individual files after ever `it` finishes.
Here I'm reusing the same file for 3 different `it` runs, so removing the file just to be safe.